lib/optimize/ConcatenatedModule.js

```diff
@@ -0,0 +1,489 @@
+							dependency: dep
+						});
+					} else if(Array.isArray(importModule.providedExports)) {
+						var activeExports = new Set(HarmonyModulesHelpers.getActiveExports(dep.originModule, dep));
+						importModule.providedExports.forEach(name => {
+							if(activeExports.has(name) || name === "default")
+								return;
+							reexportMap.set(name, {
+								module: importModule,
+								exportName: name,
+								dependency: dep
+							});
+						});
+					} else if(innerReexport) {
+						throw new Error(`Module "${importModule.readableIdentifier(requestShortener)}" doesn't provide static exports for "export *" in ${m.readableIdentifier(requestShortener)}`);
+					}
+				}
+			});
+			return {
+				module: m,
+				index: idx,
+				source: undefined,
+				globalScope: undefined,
+				moduleScope: undefined,
+				internalNames: new Map(),
+				exportMap: exportMap,
+				reexportMap: reexportMap,
+				needCompatibilityFlag: false,
+				needNamespaceObject: false
+			};
+		});
+
+		// Create mapping from module to info
+		const moduleToInfoMap = new Map();
+		modulesWithInfo.forEach(m => moduleToInfoMap.set(m.module, m));
+
+		// Configure template decorators for dependencies
+		const innerDependencyTemplates = new Map(dependencyTemplates);
+
+		innerDependencyTemplates.set(HarmonyImportSpecifierDependency, new HarmonyImportSpecifierDependencyConcatenatedTemplate(
+			dependencyTemplates.get(HarmonyImportSpecifierDependency),
+			moduleToInfoMap
+		));
+		innerDependencyTemplates.set(HarmonyImportDependency, new HarmonyImportDependencyConcatenatedTemplate(
+			dependencyTemplates.get(HarmonyImportDependency),
+			moduleToInfoMap
+		));
+		innerDependencyTemplates.set(HarmonyExportSpecifierDependency, new HarmonyExportSpecifierDependencyConcatenatedTemplate(
+			dependencyTemplates.get(HarmonyExportSpecifierDependency),
+			this.rootModule
+		));
+		innerDependencyTemplates.set(HarmonyExportExpressionDependency, new HarmonyExportExpressionDependencyConcatenatedTemplate(
+			dependencyTemplates.get(HarmonyExportExpressionDependency),
+			this.rootModule,
+			moduleToInfoMap
+		));
+		innerDependencyTemplates.set(HarmonyExportImportedSpecifierDependency, new HarmonyExportImportedSpecifierDependencyConcatenatedTemplate(
+			dependencyTemplates.get(HarmonyExportImportedSpecifierDependency),
+			this.rootModule,
+			moduleToInfoMap
+		));
+		innerDependencyTemplates.set(HarmonyCompatibilityDependency, new HarmonyCompatibilityDependencyConcatenatedTemplate(
+			dependencyTemplates.get(HarmonyCompatibilityDependency),
+			this.rootModule,
+			moduleToInfoMap
+		));
+
+		// Generate source code and analyse scopes
+		// Prepare a ReplaceSource for the final source
+		modulesWithInfo.forEach(info => {
+			const m = info.module;
+			const source = m.source(innerDependencyTemplates, outputOptions, requestShortener);
+			const code = source.source();
+			const ast = acorn.parse(code, {
+				ranges: true,
+				locations: true,
+				ecmaVersion: 2017,
```

Comment:
This probably should be a shared constant, to ensure Compiler and this share the same value